agents.py

```diff
@@ -964,24 +959,10 @@ def execute_action(self, agent, action):
 
         if isinstance(agent, Explorer) and self.in_danger(agent):
             return
-
+            
         agent.bump = False
-        if action == 'TurnRight':
-            agent.direction += Direction.R
-            agent.performance -= 1
-        elif action == 'TurnLeft':
-            agent.direction += Direction.L
-            agent.performance -= 1
-        elif action == 'Forward':
-            agent.bump = self.move_to(agent, agent.direction.move_forward(agent.location))
-            agent.performance -= 1
-        elif action == 'Grab':
-            things = [thing for thing in self.list_things_at(agent.location)
-                      if agent.can_grab(thing)]
-            if len(things):
-                print("Grabbing", things[0].__class__.__name__)
-                if len(things):
-                    agent.holding.append(things[0])
+        if action in ['TurnRight', 'TurnLeft', 'Forward', 'Grab']:
+            super().execute_action(agent,action)
```

Comment:
You need a space after the comma, to comply with the overall project style.

Reply:
done